### DISCUSSION TITLE
Wrong values for entities

### DISCUSSION THREAD

**Discussion by whaensch** [wills106/homeassistant-solax-modbus#943] [Q&A]
Hi,

I am using a Solax Mini G4 (XM4…) with PocketWiFi 3 and a DTSU666. I try to use homeassistant to read Modbus over TCP. The communication is working - but the values seam not to be correct (see Screenshot
![IMG_2854](https://github.com/wills106/homeassistant-solax-modbus/assets/109789247/34a1ca52-a426-472c-9bd3-d0e4b32ef6ae)
). 
Is it possible to correct the values?

**Comment by whaensch** [comment]
I try to integrate solax sensors in Panel energy but the interesting sensors are not present in the dialog. 
One of the sensors is sensor.solax_grid_export - but I could not choose this sensor in Energy Panel?

![IMG_2858](https://github.com/wills106/homeassistant-solax-modbus/assets/109789247/340c1e24-5e7d-4c70-a3cd-1375567ff117)
![IMG_2857](https://github.com/wills106/homeassistant-solax-modbus/assets/109789247/b2cb2511-71ef-48fa-9b77-c22cb6cfef74)

**Comment by wills106** [comment]
> Is it possible to correct the values?

Which value is incorrect? I don't own this Inverter so I can't do any testing.

Is it "CT Power" that's incorrect? Should it be 655.35? Well I think 655,35 as you use a slightly different decimal system.
You could add `scale = 0.01,`

Like "Inverter Frequency":
https://github.com/wills106/homeassistant-solax-modbus/blob/0289658890aaf568115fc249ee7775c4b9454fb5/custom_components/solax_modbus/plugin_solax.py#L6681-L6702

> One of the sensors is sensor.solax_grid_export - but I could not choose this sensor in Energy Panel?

If you replace `state_class = SensorStateClass.MEASUREMENT,` with `state_class = SensorStateClass.TOTAL_INCREASING,` on the following lines:
(They will probably be numbered different on your Version, but you should be able to search for them)
https://github.com/wills106/homeassistant-solax-modbus/blob/0289658890aaf568115fc249ee7775c4b9454fb5/custom_components/solax_modbus/plugin_solax.py#L7208

https://github.com/wills106/homeassistant-solax-modbus/blob/0289658890aaf568115fc249ee7775c4b9454fb5/custom_components/solax_modbus/plugin_solax.py#L7218

Does it allow you to add them to the dashboard?
